File: javalin-openapi-processor/src/main/java/net/dzikoysk/openapi/processor/annotations/OpenApiComposedRequestBodyInstance.java
package net.dzikoysk.openapi.processor.annotations;

import io.javalin.plugin.openapi.annotations.OpenApiComposedRequestBody;
import net.dzikoysk.openapi.processor.processing.AnnotationMirrorMapper;

import javax.lang.model.element.AnnotationMirror;
import java.lang.annotation.Annotation;

public final class OpenApiComposedRequestBodyInstance extends AnnotationMirrorMapper {

    public OpenApiComposedRequestBodyInstance(AnnotationMirror mirror) {
        super(mirror);
    }

    public OpenApiContentInstance[] anyOf() {
        return getArray("anyOf", AnnotationMirror.class, OpenApiContentInstance::new, OpenApiContentInstance[]::new);
    }

    public String contentType() {
        return getString("contentType");
    }

    public String description() {
        return getString("description");
    }

    public OpenApiContentInstance[] oneOf() {
        return getArray("oneOf", AnnotationMirror.class, OpenApiContentInstance::new, OpenApiContentInstance[]::new);
    }

    public boolean required() {
        return getBoolean("required");
    }

    public Class<? extends Annotation> annotationType() {
        return OpenApiComposedRequestBody.class;
    }

}

File: openapi-processor/src/main/kotlin/com/dzikoysk/openapi/processor/annotations/OpenApiComposedRequestBodyInstance.kt
package com.dzikoysk.openapi.processor.annotations

import com.dzikoysk.openapi.annotations.OpenApiComposedRequestBody
import com.dzikoysk.openapi.processor.processing.AnnotationMirrorMapper
import javax.lang.model.element.AnnotationMirror

internal class OpenApiComposedRequestBodyInstance(mirror: AnnotationMirror) : AnnotationMirrorMapper(mirror) {

    fun anyOf(): List<OpenApiContentInstance> =
        getArray("anyOf", AnnotationMirror::class.java) { OpenApiContentInstance(it) }

    fun contentType(): String =
        getString("contentType")

    fun description(): String =
        getString("description")

    fun oneOf(): List<OpenApiContentInstance> =
        getArray("oneOf", AnnotationMirror::class.java) { OpenApiContentInstance(it) }

    fun required(): Boolean =
        getBoolean("required")

    fun annotationType(): Class<out Annotation> =
        OpenApiComposedRequestBody::class.java

}